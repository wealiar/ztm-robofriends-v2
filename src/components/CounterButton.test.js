import { shallow } from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton';

describe('CounterButton smart component', () => {
    it('expect to render CounterButton component', () => {
        const mockColor = 'green';
        const wrapper = shallow(<CounterButton color={mockColor} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('expect to get initial value, which is 1', () => {
        const wrapper = shallow(<CounterButton />);

        expect(wrapper.text()).toBe('Count: 1');
        expect(wrapper.state()).toEqual({ count: 1 });
    });

    it('expect to click button and get incremented value, which is 2', () => {
        const wrapper = shallow(<CounterButton />);

        wrapper.simulate('click');

        expect(wrapper.text()).toBe('Count: 2');
        expect(wrapper.state()).toEqual({ count: 2 });
    });

    it('expect to update counter and value have to stay as initial', () => {
        const wrapper = shallow(<CounterButton />);
        expect(wrapper.text()).toBe('Count: 1');

        wrapper.update();
        expect(wrapper.text()).toBe('Count: 1');
    });

    it('set color using props and expect have property color with this value', () => {
        expect.assertions(4);
        const wrapper = shallow(<CounterButton color='red' />);

        expect(wrapper.prop('style')).toStrictEqual({
            color: 'red'
        });
        expect(wrapper.props().style.color).toEqual('red');

        wrapper.setProps({ 'color': 'blue' });
        wrapper.simulate('click'); // added click to re-render component
        expect(wrapper.prop('style')).toStrictEqual({
            color: 'blue'
        });
        expect(wrapper.props().style.color).toEqual('blue');

    });
});