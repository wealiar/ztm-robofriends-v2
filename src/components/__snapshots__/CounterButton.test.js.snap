// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton smart component expect to render CounterButton component 1`] = `ShallowWrapper {}`;
